src/completion.ts

```diff
@@ -442,3 +506,9 @@ export function getCompletionTriggerCharacter(character: string | undefined): ts
             return undefined;
     }
 }
+
+function rangeUnion(a: lsp.Range, b: lsp.Range): lsp.Range {
+    const start = a.start.line < b.start.line || a.start.line === b.start.line && a.start.character < b.start.character ? a.start : b.start;
+    const end = a.end.line > b.end.line || a.end.line === b.end.line && a.end.character > b.end.character ? a.end : b.end;
+    return { start, end };
+}
```

Comment:
We can move it to `src/utils/typeConverters.ts`. It already has `intersection` function.